packages/graalvm/src/main/kotlin/elide/runtime/plugins/env/EnvConfig.kt

```diff
@@ -19,9 +19,20 @@ import elide.runtime.plugins.env.EnvConfig.EnvVariableSource.INLINE
     public var enabled: Boolean = true
 
     /**
-     * Isolated suite of environment variables to provide to the guest application.
+     * Isolated suite of environment variables to provide to the guest application. Use [setEnv] to register new
+     * variable values or override existing ones.
      */
-    public val isolatedEnvironmentVariables: MutableMap<String, EnvVar> = ConcurrentSkipListMap()
+    public val isolatedEnvironmentVariables: Map<String, EnvVar> get() = mutableEnvironmentVariables
+
+    /**
+     * Private mutable map for setting environment variables.
+     */
+    private val mutableEnvironmentVariables: MutableMap<String, EnvVar> = ConcurrentSkipListMap()
+
+    /** Register an [EnvVar] with the given [key]. */
+    public fun setEnv(key: String, value: EnvVar) {
```

Comment:
Following a similar pattern to language bindings in `AbstractLanguagePlugin`, expose only an immutable map and very restricted mutating operations using methods.